arch/arm/soc/nordic_nrf/nrf52/dts.fixup

```diff
@@ -7,6 +7,10 @@
 
 #define FLASH_DEV_NAME			NRF_NRF52_FLASH_CONTROLLER_4001E000_LABEL
 
+#define CONFIG_GPIO_NRF5_P0_DEV_NAME	NORDIC_NRF5_GPIO_50000000_LABEL
+#define CONFIG_GPIO_NRF5_P1_DEV_NAME	NORDIC_NRF5_GPIO_50000300_LABEL
+#define CONFIG_GPIOTE_NRF5_PRI		NORDIC_NRF5_GPIOTE_40006000_IRQ_0_PRIORITY
```

Comment:
Since you're moving gpio to dts, I think you should get IRQ number too from dts (and remove the one from CMSIS)

Reply:
What do you mean by this? I don't see what needs to be converted in the dts.fixup

do you mean in gpio_nrf5.c, that NRF5_IRQ_GPIOTE_IRQn should be changed to NORDIC_NRF5_GPIOTE_40006000_IRQ_GPIOTE ?
```
static int gpio_nrf5_init(struct device *dev)
{
	IRQ_CONNECT(NRF5_IRQ_GPIOTE_IRQn, CONFIG_GPIOTE_NRF5_PRI,
		    gpio_nrf5_port_isr, NULL, 0);

	irq_enable(NRF5_IRQ_GPIOTE_IRQn);

	return 0;
}
```

Reply:
Indeed, but you'll need to go through a KConfig symbol, like CONFIG_GPIOTE_NRF5_IRQ

Reply:
I don't see the point

Reply:
Aim is to extract driver configuration from device tree, and use it as a reference.
Having information coming from different sources (dts and CMSIS) will be confusing at the end.
Such as defining it in device tree and not using it.
Today, there is information coming from different sources but aim is to converge to device tree.

Reply:
There should be 2 defines regarding IRQ in fixup
```
#define CONFIG_GPIOTE_NRF5_IRQ		NORDIC_NRF5_GPIOTE_40006000_IRQ_0
#define CONFIG_GPIOTE_NRF5_IRQ_PRI	NORDIC_NRF5_GPIOTE_40006000_IRQ_0_PRIORITY
```
And then change references to IRQ in gpio_nrf5.c
```
static int gpio_nrf5_init(struct device *dev)
{
	IRQ_CONNECT(CONFIG_GPIOTE_NRF5_IRQ, CONFIG_GPIOTE_NRF5_IRQ_PRI,
		    gpio_nrf5_port_isr, NULL, 0);

	irq_enable(CONFIG_GPIOTE_NRF5_IRQ_PRI);

	return 0;
}
```
This way it will be consistent with dts for other peripherals

Reply:
LGTM now 
@erwango could you please revisit?